matchzoo/datapack.py

```diff
@@ -0,0 +1,115 @@
+"""Matchzoo DataPack, pair-wise tuple (feature) and context as input."""
+
+import typing
+from pathlib import Path
+
+import dill
+import pandas as pd
+
+
+class DataPack(object):
+    """
+    Matchzoo DataPack data structure, store dataframe and context.
+
+    Example:
+        >>> features = [([1,3], [2,3]), ([3,0], [1,6])]
+        >>> context = {'vocab_size': 2000}
+        >>> dp = DataPack(data=features,
+        ...               context=context)
+        >>> type(dp.sample(1))
+        <class 'matchzoo.datapack.DataPack'>
+        >>> len(dp)
+        2
+        >>> features, context = dp.dataframe, dp.context
+        >>> context
+        {'vocab_size': 2000}
+    """
+
+    DATA_FILENAME = 'data.dill'
+
+    def __init__(self,
+                 data: list,
+                 context: dict={}):
+        """Initialize."""
+        self._dataframe = pd.DataFrame(data)
+        self._context = context
+
+    def __len__(self) -> int:
+        """Get size of the data pack."""
+        return self._dataframe.shape[0]
+
+    @property
+    def dataframe(self):
+        """Get data frame."""
+        return self._dataframe
+
+    @property
+    def context(self):
+        """Get context of `DataPack`."""
+        return self._context
+
+    def sample(self, number, replace=True):
+        """
+        Sample records from `DataPack` object, for generator.
+
+        :param number: number of records to be sampled, use `batch_size`.
+        :param replace: sample with replacement, default value is `True`.
+
+        :return data_pack: return `DataPack` object including sampled data
+                           and context (shallow copy of the context`).
+        """
+        return DataPack(self._dataframe.sample(n=number, replace=replace),
+                        self._context.copy())
+
+    def append(self, new_data_pack: 'DataPack'):
+        """
+        Append a new `DataPack` object to current `DataPack` object.
+
+        It should be noted that the context of the previous `DataPack`
+        will be updated by the new one.
+
+        :param new_data_pack: A new DataPack object.
+        :param overwite_context: Allow overwrite common context by new context.
+        """
+        new_dataframe = new_data_pack.dataframe
+        new_context = new_data_pack.context
+        self._dataframe = self._dataframe.append(
```

Comment:
Same issue. `self._dataframe.append(new_dataframe, ignore_index=True)` this is the "new_dataframe" since it's the data frame that overwrites the current one.